src/linux/udev-device-watcher.cpp

```diff
@@ -181,22 +181,21 @@ udev_device_watcher::~udev_device_watcher()
 {
     stop();
 
-    if( _udev_monitor_fd == -1 )
```

Comment:
This is not needed anymore?

Reply:
The warning was on the throw -- it said it will always terminate()
Not really needed -- I added the other ifs